### PR TITLE
GT-2430 disable share links on cyoa

### PR DESCRIPTION
**Pull request by rachaelblue**

### CODE REVIEW

File: godtools/App/Flows/ToolSettings/Observer/Protocols/LinkShareable.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//  LinkShareable.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 8/20/24.
+//  Copyright © 2024 Cru. All rights reserved.
+//
+
+import Foundation
+
+protocol LinkShareable {
```
Comment (levieggertcru) [inline godtools/App/Flows/ToolSettings/Observer/Protocols/LinkShareable.swift:11]: Hey @rachaelblue these changes are looking good.  The View fix looks good too.  

I'm wondering if we should think about adding the dependencies needed into the protocols?  Going to think about that a bit.

File: godtools/App/Features/ToolSettings/Presentation/ToolSettings/Subviews/ToolSettingsOptions/ToolSettingsOptionsView.swift [hunk 2]
```diff
@@ -28,15 +28,18 @@ struct ToolSettingsOptionsView: View {
             ScrollView(.horizontal, showsIndicators: false) {
                 HStack {
```
Comment (levieggertcru) [inline godtools/App/Features/ToolSettings/Presentation/ToolSettings/Subviews/ToolSettingsOptions/ToolSettingsOptionsView.swift:29]: I might make a separate ticket to add a ForEach here and do that later.  I should have originally done that.  We could then just check for toolOptions.isEmpty in ToolSettingsView which would be a little safer.

**Review by levieggertcru — APPROVED** [review]
Thanks @rachaelblue this looks good. I ended up making a separate ticket (https://jira.cru.org/browse/GT-2434) for adding a ForEach to the HStack for tool options.  I should've done that originally.  Not a high priority, but something that would be nice to update to in the future.

File: godtools/App/Flows/ToolSettings/Observer/Protocols/LinkShareable.swift (hunk 1 above)
Comment (aaronlaib) [reply godtools/App/Flows/ToolSettings/Observer/Protocols/LinkShareable.swift:11]: This isn't me reviewing the PR haha, but @rachaelblue I really like how you set up Tool Options, allowing easy disable and enable of the different parts.

Comment (levieggertcru) [reply godtools/App/Flows/ToolSettings/Observer/Protocols/LinkShareable.swift:11]: Hey @aaronlaib welcome.  I agree.

Comment (rachaelblue) [reply godtools/App/Flows/ToolSettings/Observer/Protocols/LinkShareable.swift:11]: Sounds good @levieggertcru.  And thanks @aaronlaib!

**Pull request merged**